lib/ecmascript.ts

```diff
@@ -221,594 +213,299 @@ function getIntlDateTimeFormatEnUsForTimeZone(timeZoneIdentifier) {
-      ({ hour, minute, second, millisecond, microsecond, nanosecond, calendar } = ES.ParseISODateTime(isoString, {
-        zoneRequired: false
-      }));
-    }
-    return { hour, minute, second, millisecond, microsecond, nanosecond, calendar };
-  },
-  ParseTemporalYearMonthString: (isoString) => {
-    const match = PARSE.yearmonth.exec(isoString);
-    let year, month, calendar, referenceISODay;
-    if (match) {
-      let yearString = match[1];
-      if (yearString[0] === '\u2212') yearString = `-${yearString.slice(1)}`;
-      year = ES.ToInteger(yearString);
-      month = ES.ToInteger(match[2]);
-      calendar = match[3];
-    } else {
-      ({ year, month, calendar, day: referenceISODay } = ES.ParseISODateTime(isoString, { zoneRequired: false }));
-    }
-    return { year, month, calendar, referenceISODay };
-  },
-  ParseTemporalMonthDayString: (isoString) => {
-    const match = PARSE.monthday.exec(isoString);
-    let month, day, calendar, referenceISOYear;
-    if (match) {
-      month = ES.ToInteger(match[1]);
-      day = ES.ToInteger(match[2]);
-    } else {
-      ({ month, day, calendar, year: referenceISOYear } = ES.ParseISODateTime(isoString, { zoneRequired: false }));
-    }
-    return { month, day, calendar, referenceISOYear };
-  },
-  ParseTemporalTimeZoneString: (stringIdent) => {
-    try {
-      let canonicalIdent = ES.GetCanonicalTimeZoneIdentifier(stringIdent);
-      if (canonicalIdent) {
-        canonicalIdent = canonicalIdent.toString();
-        if (ES.ParseOffsetString(canonicalIdent) !== null) return { offset: canonicalIdent };
-        return { ianaName: canonicalIdent };
-      }
-    } catch {
-      // fall through
+    !HasSlot(item, ISO_YEAR, ISO_MONTH, ISO_DAY)
+  );
+}
+export function IsTemporalDateTime(item) {
+  return HasSlot(
+    item,
+    ISO_YEAR,
+    ISO_MONTH,
+    ISO_DAY,
+    ISO_HOUR,
+    ISO_MINUTE,
+    ISO_SECOND,
+    ISO_MILLISECOND,
+    ISO_MICROSECOND,
+    ISO_NANOSECOND
+  );
+}
+export function IsTemporalYearMonth(item) {
+  return HasSlot(item, YEAR_MONTH_BRAND);
+}
+export function IsTemporalMonthDay(item) {
+  return HasSlot(item, MONTH_DAY_BRAND);
+}
+export function IsTemporalZonedDateTime(item) {
+  return HasSlot(item, EPOCHNANOSECONDS, TIME_ZONE, CALENDAR);
+}
+export function TemporalTimeZoneFromString(stringIdent) {
+  let { ianaName, offset, z } = ParseTemporalTimeZoneString(stringIdent);
+  let identifier = ianaName;
+  if (!identifier && z) identifier = 'UTC';
+  if (!identifier) identifier = offset;
+  const result = GetCanonicalTimeZoneIdentifier(identifier);
+  if (offset && identifier !== offset) {
+    const ns = ParseTemporalInstant(stringIdent);
+    const offsetNs = GetIANATimeZoneOffsetNanoseconds(ns, result);
+    if (FormatTimeZoneOffsetString(offsetNs) !== offset) {
+      throw new RangeError(`invalid offset ${offset}[${ianaName}]`);
     }
+  }
+  return result;
+}
+
+export function FormatCalendarAnnotation(id, showCalendar) {
```

Comment:
```suggestion
export function FormatCalendarAnnotation(id: string, showCalendar: 'auto' | 'never') {
```

At some point the arguments to all exported ecmascript.js methods should probably get type annotations. Is your plan to do this in a follow-up PR?  Or would it make sense to do it in this one? I don't have an opinion either way, just wondering.